### PR TITLE
x-pack/filebeat/input/entityanalytics/provider/okta: new package

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds the input package for the Okta identity provider.

## Why is it important?

Roadmap issue.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation ~**TODO**~
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- For elastic/security-team#5100
- Depends #35478

### CODE REVIEW

**Review by taylor-swanson — COMMENTED** [review]
I see that there is still a TODO in the checklist, do you have any other changes pending?

Otherwise, just a minor comment and fixing the linter issues is all I've got right now. Changes LGTM and will approve once I know there are no other outstanding items.

File: x-pack/filebeat/input/entityanalytics/provider/okta/okta.go [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+			}
+			p.metrics.updateTotal.Inc()
+			p.metrics.updateProcessingTime.Update(time.Since(start).Nanoseconds())
+			updateTimer.Reset(p.cfg.UpdateInterval)
+			p.logger.Debugf("Next update expected at: %v", time.Now().Add(p.cfg.UpdateInterval))
+		}
+	}
+}
+
+func newClient(ctx context.Context, cfg conf, log *logp.Logger) (*http.Client, error) {
```
[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/okta/okta.go:156]: Removed.

**Comment by efd6** [comment]
The TODO was TODONE in the second commit, but I neglected to update. I'd like to wait on a code client, then I'll ping here when that's done.

File: x-pack/filebeat/input/entityanalytics/provider/okta/conf_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		},
+		wantErr: errSyncBeforeUpdate,
+	},
+}
+
+func TestConfValidate(t *testing.T) {
+	for _, test := range validateTests {
+		t.Run(test.name, func(t *testing.T) {
+			err := test.cfg.Validate()
+			if err != test.wantErr {
```
Comment (efd6) [inline x-pack/filebeat/input/entityanalytics/provider/okta/conf_test.go:57]: This linter warning should be ignored. The errors are not wrapped and this explicitly tests the errors as part of testing.

**Review by ShourieG — APPROVED** [review]
LGTM

**Pull request merged**